addons/pos_self_order/static/src/app/pages/cart_page/cart_page.js

```diff
@@ -53,6 +53,17 @@ export class CartPage extends Component {
             return;
         }
 
+        if (
+            this.selfOrder.currentTable &&
+            !this.selfOrder.currentOrder.table_id &&
+            type === "mobile" &&
+            orderingMode === "table"
+        ) {
+            this.selfOrder.currentOrder.update({
+                table_id: this.selfOrder.currentTable,
+            });
+        }
+
```

Comment:
You could combine this if with the next one to not duplicate condition verification